File: lib/spiral/date_time_wasm.py
from collections.abc import Callable
from typing import Any
from fable_modules.fable_library.date import (to_string, parse_exact)
from fable_modules.fable_library.guid import (parse, new_guid)
from fable_modules.fable_library.long import parse as parse_1
from fable_modules.fable_library.option import value
from fable_modules.fable_library.string_ import (replace, pad_left)
from fable_modules.fable_library.types import int64
from fable_modules.fable_library.util import int64_to_string

def closure1(v0_1: str, v1_1: Any) -> str:
    v3_1: str = str(v0_1)
    v4_1: str = to_string(v1_1, "yyyyMMdd-HHmm-ssff-ffff-f")
    return parse(((("" + v4_1) + "") + v3_1[len(v4_1):len(v3_1)]) + "")


def closure0(unit_var: None, v0_1: str) -> Callable[[Any], str]:
    def _arrow24(v: Any, unit_var: Any=unit_var, v0_1: Any=v0_1) -> str:
        return closure1(v0_1, v)

    return _arrow24


def closure5(v0_1: str, v1_1: str, v2_1: str) -> str:
    return replace(v2_1, v0_1, v1_1)


def closure4(v0_1: str, v1_1: str) -> Callable[[str], str]:
    def _arrow25(v: str, v0_1: Any=v0_1, v1_1: Any=v1_1) -> str:
        return closure5(v0_1, v1_1, v)

    return _arrow25


def closure3(unit_var: None, v0_1: str) -> Callable[[str, str], str]:
    def _arrow26(v: str, unit_var: Any=unit_var, v0_1: Any=v0_1) -> Callable[[str], str]:
        return closure4(v0_1, v)

    return _arrow26


def method0(__unit: None=None) -> Callable[[str, str, str], str]:
    def _arrow27(v: str) -> Callable[[str, str], str]:
        return closure3(None, v)

    return _arrow27


def closure2(unit_var: None, v0_1: str) -> Any:
    v2_1: str = str(v0_1)
    _v4: Any | None = None
    x: Any = parse_exact(method0()("-")("")(v2_1[:24 + 1]), "yyyyMMddHHmmssfffffff", None)
    _v4 = x
    return value(_v4)


def closure7(v0_1: str, v1_1: int64) -> str:
    v3_1: str = str(v0_1)
    v6_1: str = pad_left(int64_to_string(v1_1), 18, "0")
    return parse(((((((((("" + v6_1[0:7 + 1]) + "-") + v6_1[8:11 + 1]) + "-") + v6_1[12:15 + 1]) + "-") + v6_1[16:17 + 1]) + "") + v3_1[21:len(v3_1)]) + "")


def closure6(unit_var: None, v0_1: str) -> Callable[[int64], str]:
    def _arrow29(v: int64, unit_var: Any=unit_var, v0_1: Any=v0_1) -> str:
        return closure7(v0_1, v)

    return _arrow29


def closure8(unit_var: None, v0_1: str) -> int64:
    v2_1: str = str(v0_1)
    return parse_1(((((((("" + v2_1[0:7 + 1]) + "") + v2_1[9:12 + 1]) + "") + v2_1[14:17 + 1]) + "") + v2_1[19:20 + 1]) + "", 511, False, 64)


def closure9(unit_var: None, v0_1: Any) -> str:
    v3_1: str = str(new_guid())
    v4_1: str = to_string(v0_1, "yyyyMMdd-HHmm-ssff-ffff-f")
    return parse(((("" + v4_1) + "") + v3_1[len(v4_1):len(v3_1)]) + "")


def closure10(unit_var: None, v0_1: int64) -> str:
    v3_1: str = str(new_guid())
    v6_1: str = pad_left(int64_to_string(v0_1), 18, "0")
    return parse(((((((((("" + v6_1[0:7 + 1]) + "-") + v6_1[8:11 + 1]) + "-") + v6_1[12:15 + 1]) + "-") + v6_1[16:17 + 1]) + "") + v3_1[21:len(v3_1)]) + "")


def closure12(v0_1: str, v1_1: Any) -> str:
    return to_string(v1_1, v0_1)


def closure11(unit_var: None, v0_1: str) -> Callable[[Any], str]:
    def _arrow31(v: Any, unit_var: Any=unit_var, v0_1: Any=v0_1) -> str:
        return closure12(v0_1, v)

    return _arrow31


def closure13(unit_var: None, v0_1: Any) -> str:
    return to_string(v0_1, "yyyy-MM-ddTHH-mm-ss.fff")


def _arrow32(v: str) -> Callable[[Any], str]:
    return closure0(None, v)


v0: Callable[[str, Any], str] = _arrow32

def date_time_guid_from_date_time(x: str) -> Callable[[Any], str]:
    return v0(x)


def _arrow33(v: str) -> Any:
    return closure2(None, v)


v1: Callable[[str], Any] = _arrow33

def date_time_from_guid(x: str) -> Any:
    return v1(x)


def _arrow35(v: str) -> Callable[[int64], str]:
    return closure6(None, v)


v2: Callable[[str, int64], str] = _arrow35

def timestamp_guid_from_timestamp(x: str) -> Callable[[int64], str]:
    return v2(x)


def _arrow36(v: str) -> int64:
    return closure8(None, v)


v3: Callable[[str], int64] = _arrow36

def timestamp_from_guid(x: str) -> int64:
    return v3(x)


def _arrow37(v: Any) -> str:
    return closure9(None, v)


v4: Callable[[Any], str] = _arrow37

def new_guid_from_date_time(x: Any) -> str:
    return v4(x)


def _arrow38(v: int64) -> str:
    return closure10(None, v)


v5: Callable[[int64], str] = _arrow38

def new_guid_from_timestamp(x: int64) -> str:
    return v5(x)


def _arrow40(v: str) -> Callable[[Any], str]:
    return closure11(None, v)


v6: Callable[[str, Any], str] = _arrow40

def format(x: str) -> Callable[[Any], str]:
    return v6(x)


def _arrow41(v: Any) -> str:
    return closure13(None, v)


v7: Callable[[Any], str] = _arrow41

def format_iso8601(x: Any) -> str:
    return v7(x)

File: lib/spiral/date_time_wasm.php
<?php
namespace Date_time;

require_once(__FABLE_LIBRARY__.'/BigInt.php');
require_once(__FABLE_LIBRARY__.'/Date.php');
require_once(__FABLE_LIBRARY__.'/Guid.php');
require_once(__FABLE_LIBRARY__.'/Long.php');
require_once(__FABLE_LIBRARY__.'/Option.php');
require_once(__FABLE_LIBRARY__.'/String.php');
require_once(__FABLE_LIBRARY__.'/Util.php');

#0
function closure1($v0_1, $v1_1) {
    $v3_1 = $v0_1;
    $v4_1 = $v1_1->\Date\toString('yyyyMMdd-HHmm-ssff-ffff-f');
    return \Guid\parse(NULL);
}

#1
function closure0($unitVar, $v0_1) {
    return function ($v) use ($closure1, $v0_1) {     return closure1($v0_1, $v);
 };
}

#2
function closure5($v0_1, $v1_1, $v2_1) {
    return $v2_1->\String\replace($v0_1, $v1_1);
}

#3
function closure4($v0_1, $v1_1) {
    return function ($v) use ($closure5, $v0_1, $v1_1) {     return closure5($v0_1, $v1_1, $v);
 };
}

#4
function closure3($unitVar, $v0_1) {
    return function ($v) use ($closure4, $v0_1) {     return closure4($v0_1, $v);
 };
}

#5
function method0($unitVar) {
    return function ($v) use ($closure3) {     return closure3(NULL, $v);
 };
}

#6
function closure2($unitVar, $v0_1) {
    $v3_1 = method0(NULL);
    $_v4 = NULL;
    $x = \Date\minValue();
    $_v4 = $x;
    return \Option\value($_v4);
}

#7
function closure7($v0_1, $v1_1) {
    $v3_1 = $v0_1;
    $v6_1 = \Util\int64ToString($v1_1)->\String\padLeft(18, '0');
    return \Guid\parse(NULL);
}

#8
function closure6($unitVar, $v0_1) {
    return function ($v) use ($closure7, $v0_1) {     return closure7($v0_1, $v);
 };
}

#9
function closure8($unitVar, $v0_1) {
    $v2_1 = $v0_1;
    return \BigInt\toInt64(\Long\parse(NULL, 511, false, 64));
}

#10
function closure9($unitVar, $v0_1) {
    $v3_1 = \Guid\newGuid();
    $v4_1 = $v0_1->\Date\toString('yyyyMMdd-HHmm-ssff-ffff-f');
    return \Guid\parse(NULL);
}

#11
function closure10($unitVar, $v0_1) {
    $v3_1 = \Guid\newGuid();
    $v6_1 = \Util\int64ToString($v0_1)->\String\padLeft(18, '0');
    return \Guid\parse(NULL);
}

#12
function closure12($v0_1, $v1_1) {
    return $v1_1->\Date\toString($v0_1);
}

#13
function closure11($unitVar, $v0_1) {
    return function ($v) use ($closure12, $v0_1) {     return closure12($v0_1, $v);
 };
}

#14
function closure13($unitVar, $v0_1) {
    return $v0_1->\Date\toString('yyyy-MM-ddTHH-mm-ss.fff');
}

#15
$GLOBALS['v0'] = function ($v) use ($closure0) { return closure0(NULL, $v);
 };

#16
function date_time_guid_from_date_time($x) {
    return $GLOBALS['v0']($x);
}

#17
$GLOBALS['v1'] = function ($v) use ($closure2) { return closure2(NULL, $v);
 };

#18
function date_time_from_guid($x) {
    return $GLOBALS['v1']($x);
}

#19
$GLOBALS['v2'] = function ($v) use ($closure6) { return closure6(NULL, $v);
 };

#20
function timestamp_guid_from_timestamp($x) {
    return $GLOBALS['v2']($x);
}

#21
$GLOBALS['v3'] = function ($v) use ($closure8) { return closure8(NULL, $v);
 };

#22
function timestamp_from_guid($x) {
    return $GLOBALS['v3']($x);
}

#23
$GLOBALS['v4'] = function ($v) use ($closure9) { return closure9(NULL, $v);
 };

#24
function new_guid_from_date_time($x) {
    return $GLOBALS['v4']($x);
}

#25
$GLOBALS['v5'] = function ($v) use ($closure10) { return closure10(NULL, $v);
 };

#26
function new_guid_from_timestamp($x) {
    return $GLOBALS['v5']($x);
}

#27
$GLOBALS['v6'] = function ($v) use ($closure11) { return closure11(NULL, $v);
 };

#28
function format($x) {
    return $GLOBALS['v6']($x);
}

#29
$GLOBALS['v7'] = function ($v) use ($closure13) { return closure13(NULL, $v);
 };

#30
function format_iso8601($x) {
    return $GLOBALS['v7']($x);
}

#31

